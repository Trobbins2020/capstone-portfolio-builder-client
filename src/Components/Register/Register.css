.RegistrationForm {
    font-family: 'Roboto', sans-serif;
    max-width: 600px;
    margin: 10% auto;
    text-align: center;
    padding: 10px;
    border-radius: 15px;
    box-shadow: -1px -1px 12px 5px rgba(232, 232, 232, 1);
}

.inputDiv {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 10px 130px;
}

.inputDiv input {
    padding: 10px;
}

.RegistrationForm input[type=submit] {
    outline: none;
    padding: 5px 15px;
    color: white;
    background-color: rgb(42, 176, 230);
    border: 2px white solid;
    box-shadow: -1px -1px 12px 3px rgb(42, 176, 230);
    font-size: 15px;
    border-radius: 30px;
}

.RegistrationForm input[type="submit"]:hover {
    color: rgb(42, 176, 230);
    background-color: white;
}

.col_theme {
    color: rgb(42, 176, 230);
}


.mb-3 {
    margin-bottom: 30px;
}

.mt-3 {
    margin-top: 30px;
}

.mb-2{
    margin-bottom: 20px;
}