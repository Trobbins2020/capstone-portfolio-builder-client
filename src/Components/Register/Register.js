import React, { Component } from "react";
import { Link } from "react-router-dom";
import AuthApiService from "../../services/auth-api-service";
import "./Register.css";
export default class Register extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {},
  };
  state = { error: null };

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { user_name, password } = ev.target;

    this.setState({ error: null });
    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value,
    })
      .then((user) => {
        user_name.value = "";
        password.value = "";
        this.props.onRegistrationSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <div>
        <form className="RegistrationForm" onSubmit={this.handleSubmit}>
          <h2 className=" mb-3 mt-3">
            <Link to="/">Portfolio Builder</Link>
          </h2>
          <h1 className="mb-3 mt-3 col_theme">Register</h1>
          <div className="error-message">
            <p>{error}</p>
          </div>
          <div className="inputDiv">
            <label htmlFor="user_name">User </label>
            <input type="text" name="user_name" id="username" required />
          </div>
          <div className="inputDiv">
            <label htmlFor="password">Password</label>
            <input
              type="password"
              name="password"
              id="password"
              required
              autoComplete="on"
            />
          </div>
          <div className="text-center mb-3 mt-3">
            <Link to="/login" className="col_theme">
              Log in Instead
            </Link>
          </div>
          <div className="text-center mb-2">
            <input type="submit" value="Submit" />
          </div>
        </form>
      </div>
    );
  }
}
