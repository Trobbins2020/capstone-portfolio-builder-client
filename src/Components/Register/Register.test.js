import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Register from "./Register";

describe(`Register component`, () => {
  it("renders the Component", () => {
    const wrapper = shallow(<Register />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
