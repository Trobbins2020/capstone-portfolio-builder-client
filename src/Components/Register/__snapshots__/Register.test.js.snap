// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders the Component 1`] = `
<div>
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <h1
      className=" mb-3 mt-3"
    >
      <Link
        replace={false}
        to="/"
      >
        Portfolio Builder
      </Link>
    </h1>
    <h1
      className="mb-3 mt-3 col_theme"
    >
      Register
    </h1>
    <div
      className="error-message"
    />
    <div
      className="inputdiv"
    >
      <label
        htmlFor="user_name"
      >
        User 
      </label>
      <input
        id="username"
        name="user_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="inputdiv"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="text-center mb-3 mt-3"
    >
      <Link
        className="col_theme"
        replace={false}
        to="/login"
      >
        Log in Instead
      </Link>
    </div>
    <div
      className="text-center mb-2"
    >
      <input
        type="submit"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
